core/http-auth/src/main/java/software/amazon/awssdk/http/auth/AwsV4HttpSigner.java

```diff
@@ -82,6 +83,47 @@ public interface AwsV4HttpSigner extends HttpSigner<AwsCredentialsIdentity> {
     SignerProperty<Boolean> NORMALIZE_PATH =
         SignerProperty.create(Boolean.class, "NormalizePath");
 
+    /**
+     * The location where auth-related data is inserted, as a result of signing.
+     * The valid choices are "Header" and "QueryString", where "Header" indicates that request
+     * headers are added, and "QueryString" indicates query-parameters are added.
+     * This property defaults to "Header".
+     */
+    SignerProperty<String> AUTH_LOCATION =
+        SignerProperty.create(String.class, "AuthLocation");
+
+    /**
+     * The duration for the request to be valid.
+     * This property defaults to the max valid duration (7 days).
+     * This is only used in the case of a pre-signing implementation.
+     */
+    SignerProperty<Duration> EXPIRATION_DURATION =
+        SignerProperty.create(Duration.class, "ExpirationDuration");
+
+    /**
+     * Whether to indicate that a payload is signed or not.
+     * This property defaults to true.
+     * This can be set false to disable payload signing.
+     */
+    SignerProperty<Boolean> PAYLOAD_SIGNING =
+        SignerProperty.create(Boolean.class, "PayloadSigning");
+
+    /**
+     * Whether to handle a request as an event-streaming or not.
+     * This property defaults to false.
+     * This is only used in the case of an implementation that supports event-streams.
+     */
+    SignerProperty<Boolean> EVENT_STREAMING =
+        SignerProperty.create(Boolean.class, "EventStreaming");
+
+    /**
+     * Whether to enable chunked encoding or not.
+     * This property defaults to false.
+     * This is only used in the case of an implementation that supports chunked-encoding.
+     */
+    SignerProperty<Boolean> CHUNKED_ENCODING =
```

Comment:
The same thought as above may apply to this. Is chunked encoding considered a property of the request and how chunks are signed is behavior of the signer based on if PayloadSigning=true?

Reply:
Whether chunked encoding is used is determined in a complex way. Essentially, we use chunked encoding if we need to. We need to depending on a combination of factors determined by the SDK. Those factors include whether we're compressing (therefore, we don't know the length), whether we're doing trailing checksums, etc.

It's surprisingly difficult to tell by the model. It's likely better to not try to couple this one to the model, and just let the signer check the request to see if something upstream decided it should be chunked.